src/Processors/Transforms/SquashingTransform.h

```diff
@@ -24,28 +24,27 @@ class SquashingChunksTransform : public ExceptionKeepingTransform
     void onFinish() override;
 
 private:
-    SquashingTransform squashing;
+    Squashing squashing;
     Chunk cur_chunk;
     Chunk finish_chunk;
 };
 
 /// Doesn't care about propagating exceptions and thus doesn't throw LOGICAL_ERROR if the following transform closes its input port.
-class SimpleSquashingChunksTransform : public IInflatingTransform
+class SimpleSquashingTransform : public ISimpleTransform
```

Comment:
Why is it rewritten back to `ISimpleTransform` after https://github.com/ClickHouse/ClickHouse/pull/63691 ?

Implementation based on `ISimpleTransform` still has a bug which lead to lost chunks, e.g. https://s3.amazonaws.com/clickhouse-test-reports/0/e2cb17ca57124545602a40a1380f3dbae6517347/fast_test.html

(quite easy to reproduce locally just running `00857_global_joinsavel_table_alias` in release build in a loop)

@CheSema

Reply:
Ref https://github.com/ClickHouse/ClickHouse/pull/66174

Reply:
As far as I understood, this test was constantly failing a long time before this PR: [link](https://play.clickhouse.com/play?user=play#U0VMRUNUIGNoZWNrX3N0YXJ0X3RpbWUsIHB1bGxfcmVxdWVzdF9udW1iZXIsIGNoZWNrX25hbWUsIHRlc3RfbmFtZSwgcmVwb3J0X3VybApGUk9NIGNoZWNrcwpXSEVSRSBjaGVja19zdGFydF90aW1lID49IG5vdygpIC0gSU5URVJWQUwgMyBNT05USAogICAgQU5EIHRlc3Rfc3RhdHVzICE9ICdTS0lQUEVEJwogICAgQU5EIHRlc3Rfc3RhdHVzIExJS0UgJ0YlJwogICAgQU5EIGNoZWNrX3N0YXR1cyAhPSAnc3VjY2VzcycKICAgIEFORCBwb3NpdGlvbih0ZXN0X25hbWUsICcwMDg1N19nbG9iYWxfam9pbnNhdmVsX3RhYmxlX2FsaWFzJykgPiAwCk9SREVSIEJZIGNoZWNrX3N0YXJ0X3RpbWU=)

So I don't know if my changes were causing failures and it will be interesting to investigate the definite reason for failures.

Reply:
It was fixed in https://github.com/ClickHouse/ClickHouse/pull/63691, which was merged on May 23.

I rechecked the failures between May 23 and June 19 (this PR is merged), these are from backport or release PRs, or simply broken commits in other PRs.